### PR TITLE
Add a workflow template for spell checking (cspell-action)

### PR DESCRIPTION
**Pull request by i-oden**
### Why

When working on a task, preparing is for review or reviewing someone elses work, it's common that a part of it consists of correcting spelling mistakes. 

Using an automatic spelling check would simplify this, reduce trivial review comments, and save time and energy for everyone. 

For this reason, this PR introduces a workflow template: Spell checking using CSpell (`cspell-action`)

### Included in PR

- New directories: 
  - `.github` and subdirectory `.github/workflows`
    - Called "workflow directory" below
  - `.config` and subdirectory `.config/cspell`
    - Called "config directory" below
    - Allows us to have multiple configurations for different purposes while still keeping the repository structured and easy to follow 
- New files: 
  - `spellcheck.yml` in workflow directory
    - The actual workflow template
    - Runs the spellcheck when a PR is opened or updated
    - Scans all types of files
    - Only scans files included in the PR diff
  - `cspell-config.yml` in config directory
    - CSpell configuration. Used by `spellcheck.yml` file during run
  - `project-specific-words.txt` in config directory
    - List of words that CSpell should **not** flag. The examples are from this repo and the 
`development_guidelines` repo that were not included in any available CSpell dictionary (that I could find) 
  - `README.md` in the config directory
    - Detailed information on current configuration and how to use the workflow and related configuration files 

### Related issues / tasks

Technical Enablement board, task [TE-91](https://scilifelab.atlassian.net/jira/software/projects/TE/boards/495?selectedIssue=TE-91)

The task was started as a result of working on the [PR guidelines](https://github.com/ScilifelabDataCentre/development-guidelines/pull/54) and the [PR template](https://github.com/ScilifelabDataCentre/data-centre-template/pull/3).

### Notes for reviewers

When reviewing, please especially focus on checking whether the README is understandable. A good way to review is to try to incorporate CSpell into your repo. When reading the README, go to the branch and view the rendered file, not in plain markdown -- it includes a flowchart that needs to be rendered to make sense. 

When spelling issues are flagged, this is what it looks like in the diff view: 
test-cspell-svenska
test-cspell-english
test-cspell-projspec

## For the author

The following items should be checked **before** marking the PR as **Ready for Review**.
For checklist items that do not apply, add a short note explaining why.

### PR context

- [x] I've written an informative title, finishing the sentence "This PR will..."
- [x] I've explained [what this PR does and why](#description)
- [x] I've mentioned and linked [related issues / tasks](#related-issues--tasks)
- [x] I've documented breaking changes, risks, assumptions, or open questions, if any

### [Definition of Done](#LINK-TO-MERGED-GUIDELINES)

- [x] The branch is up to date with `<default-branch>`
- [x] The PR aligns with the [Acceptance Criteria](#LINK-TO-MERGED-GUIDELINES)
- [x] All configured checks pass
- [x] Tests confirm the intended behaviour - Tests in this case include the screenshots above.
- [x] Unused code and debug leftovers have been removed
- [x] Naming and structure follow the agreed [style guidelines](#LINK-TO-STYLE-GUIDELINES)
- [x] The changes made in the PR are documented appropriately

### Self-review

- [x] I've performed a [self-review](#LINK-TO-MERGED-GUIDELINES) of this PR

## For the reviewer

See the guidelines on [how to review a PR](#LINK-TO-MERGED-GUIDELINES). They offer useful tips that simplify the process for everyone involved.

Use the following checklist during the review.

Review broad to narrow:

- [ ] I understand the purpose of the PR
- [ ] The scope of the PR matches the [Acceptance Criteria](#LINK-TO-MERGED-GUIDELINES)
- [ ] The behaviour matches the description
- [ ] Risks, assumptions, edge cases, and possible side effects have been considered
- [ ] Tests are present and meaningful, or the absence of tests is acceptable
- [ ] The code is clean, readable, maintainable, and follows agreed standards

Once the PR aligns with the Definition of Done and any discussions are resolved, approve the PR.

[TE-91]: https://scilifelab.atlassian.net/browse/TE-91?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by i-oden** [comment]
@codex review

File: .github/workflows/spellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      # Checkout the repo to access the files for spell checking
+      - name: Checkout Repository
+        uses: actions/checkout@v6
+        with:
+          persist-credentials: false # We don't need the token in subsequent steps
+
+      # Some dictionaries require installation here
+      # and import in the .config/cspell/cspell-config.yml file
+      - name: Install CSpell Dictionaries (Swedish and names)
+        run: npm install --no-save @cspell/dict-sv @cspell/dict-people-names
```
[the remark this answers is not in the document]
Comment (i-oden) [reply .github/workflows/spellcheck.yml]: @codex review again

**Comment by i-oden** [comment]
@codex review

**Comment by i-oden** [comment]
@codex review
